### PR TITLE
put ADFS config artifacts under revision control (SOC-11005)

### PR DESCRIPTION
**Pull request by guangyee**
We should advise customers to put adfs_config.yaml, adfs_medatada.xml,
and adfs_mappings.json under revision control, just like other Ardana
configuration files so that we can sanely keep track of them.

### CODE REVIEW

File: xml/operations-configuring-identity-websso.xml [hunk 1]
```diff
@@ -356,7 +358,7 @@ keystone_sp_conf:
         description: This is the ADFS identity provider.
     mapping:
         id: mapping1
-        rules_file: adfs_mapping.json
+        rules_file: /var/lib/ardana/openstack/my_cloud/config/adfs_mapping.json
```
Comment (mbelur) [inline xml/operations-configuring-identity-websso.xml]: Shouldn't there be a keystone folder in the path?

Comment (guangyee) [reply xml/operations-configuring-identity-websso.xml]: ahhh good catch!

File: xml/operations-configuring-identity-websso.xml [hunk 2]
```diff
@@ -322,22 +322,24 @@
...
    <listitem>
     <para>
-     Copy adfs_metadata.xml to the &clm; node in your preferred
-     location. Here it is /tmp.
+     Copy adfs_metadata.xml to the &clm; node and place it into <filename>/var/lib/ardana/openstack/my_cloud/config/keystone/</filename> directory and put it under revision control.
     </para>
+<screen>&prompt.ardana;cd ~/openstack
+&prompt.ardana;git checkout site
+&prompt.ardana;git add my_cloud/config/keystone/adfs_metadata.xml
+&prompt.ardana;git commit -m "Add ADFS metadata file for WebSSO authentication"
```
Comment (jeremy-moffitt) [inline xml/operations-configuring-identity-websso.xml]: think you're missing the `</screen>` closing tag here

File: xml/operations-configuring-identity-websso.xml [hunk 3]
```diff
@@ -446,12 +442,22 @@ keystone_sp_conf:
...
    </listitem>
+   <listitem>
+    <para>
+     Put both adfs_config.yml and adfs_mapping.json under revision control.
+    </para>
+<screen>&prompt.ardana;cd ~/openstack
+&prompt.ardana;git checkout site
+&prompt.ardana;git add my_cloud/config/keystone/adfs_config.yml
+&prompt.ardana;git add my_cloud/config/keystone/adfs_mapping.json
+&prompt.ardana;git commit -m "Add ADFS config and mapping."
```
Comment (jeremy-moffitt) [inline xml/operations-configuring-identity-websso.xml]: missing closing tag `</screen>`

File: xml/operations-configuring-identity-websso.xml (hunk 2 above)
Comment (guangyee) [reply xml/operations-configuring-identity-websso.xml]: good catch!

File: xml/operations-configuring-identity-websso.xml (hunk 3 above)
Comment (guangyee) [reply xml/operations-configuring-identity-websso.xml]: good catch!

File: xml/operations-configuring-identity-websso.xml [hunk 4]
```diff
@@ -322,22 +322,25 @@
    </listitem>
    <listitem>
     <para>
-     Copy adfs_metadata.xml to the &clm; node in your preferred
-     location. Here it is /tmp.
+     Copy adfs_metadata.xml to the &clm; node and place it into <filename>/var/lib/ardana/openstack/my_cloud/config/keystone/</filename> directory and put it under revision control.
```
Comment (asettle) [inline xml/operations-configuring-identity-websso.xml]: Could you please wrap this text to make it easier to review? :)

File: xml/operations-configuring-identity-websso.xml [hunk 5]
```diff
@@ -322,22 +322,25 @@
...
    </listitem>
   </orderedlist>
  </section>
  <section>
   <title>Setting Up WebSSO</title>
   <para>
    Start by creating a config file <filename>adfs_config.yml</filename> with the
-   following parameters and place it in any directory on your &clm;,
-   such as <filename>/tmp</filename>.
+   following parameters and place it in the <filename>/var/lib/ardana/openstack/my_cloud/config/keystone/</filename> directory on your &clm; node.
```
Comment (asettle) [inline xml/operations-configuring-identity-websso.xml]: Same here please

File: xml/operations-configuring-identity-websso.xml [hunk 6]
```diff
@@ -402,16 +405,10 @@ keystone_sp_conf:
...
-     idp_metadata_file references the previously generated ADFS metadata file.
-     In this case:
-    </para>
-<screen>idp_metadata_file: /tmp/adfs_metadata.xml</screen>
-   </listitem>
-   <listitem>
-    <para>
-     Create a mapping file that is referenced from the preceding config file,
-     such as /tmp/adfs_sp_mapping.json. rules_file: /tmp/adfs_sp_mapping.json.
+     Create a mapping file, adfs_mapping.json, that is referenced from the
```
Comment (asettle) [inline xml/operations-configuring-identity-websso.xml]:
```suggestion
     Create a mapping file, <filename>adfs_mapping.json</filename>, that is referenced from the

File: xml/operations-configuring-identity-websso.xml [hunk 7]
```diff
@@ -402,16 +405,10 @@ keystone_sp_conf:
...
+     preceding config file in
+     <filename>/var/lib/ardana/openstack/my_cloud/config/keystone/</filename>.
+     rules_file: /var/lib/ardana/openstack/my_cloud/config/keystone/adfs_mapping.json.
```
Comment (asettle) [inline xml/operations-configuring-identity-websso.xml]: What's this second line mean?

File: xml/operations-configuring-identity-websso.xml [hunk 8]
```diff
@@ -446,12 +443,23 @@ keystone_sp_conf:
      Also see <xref linkend="maprules"/> for more information.
     </para>
    </listitem>
+   <listitem>
+    <para>
+     Put both adfs_config.yml and adfs_mapping.json under revision control.
```
Comment (asettle) [inline xml/operations-configuring-identity-websso.xml]:
```suggestion
     Add <filename>adfs_config.yml</filename> and <filename>adfs_mapping.json</filename>
     to revision control:

File: xml/operations-configuring-identity-websso.xml (hunk 4 above)
Comment (guangyee) [reply xml/operations-configuring-identity-websso.xml]: Done

File: xml/operations-configuring-identity-websso.xml (hunk 5 above)
Comment (guangyee) [reply xml/operations-configuring-identity-websso.xml]: Done

File: xml/operations-configuring-identity-websso.xml (hunk 6 above)
Comment (guangyee) [reply xml/operations-configuring-identity-websso.xml]: Done

File: xml/operations-configuring-identity-websso.xml (hunk 7 above)
Comment (guangyee) [reply xml/operations-configuring-identity-websso.xml]: That's an example of the rules_file line. Maybe I need to encapsulate it in <screen>?

File: xml/operations-configuring-identity-websso.xml (hunk 8 above)
Comment (guangyee) [reply xml/operations-configuring-identity-websso.xml]: Done

**Comment by guangyee** [comment]
> We should add the commands to run the config-processor-run.yml and ready-deployment.yml playbooks after the git commit with the websso artefacts?

But we are not changing the input model though, so I don't think we need to run those playbooks. We are merely putting the ADFS artifacts under version control.

File: xml/operations-configuring-identity-websso.xml (hunk 7 above)
Comment (asettle) [reply xml/operations-configuring-identity-websso.xml]: Yeah :) it would be helpful to be marked up.

Comment (guangyee) [reply xml/operations-configuring-identity-websso.xml]: Done. Put it in <screen>.

**Pull request merged**